### ISSUE TITLE
Hue 3.8.0 - Fork in workflow is not working as expected.

### ISSUE THREAD

**Issue by searchwizard** [cloudera/hue#181]
When I drag and drop an action next to another action, its adding a fork node in the workflow but the links are not drawn properly. I can manually fix the "OK" of the previous action to go to the fork node. But the link from fork can't be adjusted. 

![image](https://cloud.githubusercontent.com/assets/6089654/7378006/58ca3dc2-edb8-11e4-8685-6eba08b15343.png)

If I submit with out fixing the workflow, I get the following error
```
File "/opt/hue/build/env/lib/python2.6/site-packages/Django-1.6.10-py2.6.egg/django/core/handlers/base.py", line 112, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)

  File "/opt/hue/build/env/lib/python2.6/site-packages/Django-1.6.10-py2.6.egg/django/db/transaction.py", line 371, in inner
    return func(*args, **kwargs)

  File "/opt/hue/apps/oozie/src/oozie/decorators.py", line 60, in decorate
    return view_func(request, *args, **kwargs)

  File "/opt/hue/apps/oozie/src/oozie/views/editor2.py", line 351, in submit_workflow
    job_id = _submit_workflow(request.user, request.fs, request.jt, workflow, mapping)

  File "/opt/hue/apps/oozie/src/oozie/views/editor2.py", line 373, in _submit_workflow
    job_id = submission.run()

  File "/opt/hue/desktop/libs/liboozie/src/liboozie/submission2.py", line 85, in run
    deployment_dir = self.deploy()

  File "/opt/hue/desktop/libs/liboozie/src/liboozie/submission2.py", line 166, in deploy
    oozie_xml = self.job.to_xml(self.properties)

  File "/opt/hue/apps/oozie/src/oozie/models2.py", line 188, in to_xml
    'workflow_mapping': workflow_mapping

  File "/opt/hue/desktop/core/src/desktop/lib/django_mako.py", line 114, in render_to_string_normal
    result = template.render(**data_dict)

  File "/opt/hue/build/env/lib/python2.6/site-packages/Mako-0.8.1-py2.6.egg/mako/template.py", line 443, in render
    return runtime._render(self, self.callable_, args, data)

  File "/opt/hue/build/env/lib/python2.6/site-packages/Mako-0.8.1-py2.6.egg/mako/runtime.py", line 786, in _render
    **_kwargs_for_callable(callable_, data))

  File "/opt/hue/build/env/lib/python2.6/site-packages/Mako-0.8.1-py2.6.egg/mako/runtime.py", line 818, in _render_context
    _exec_template(inherit, lclcontext, args=args, kwargs=kwargs)

  File "/opt/hue/build/env/lib/python2.6/site-packages/Mako-0.8.1-py2.6.egg/mako/runtime.py", line 844, in _exec_template
    callable_(context, *args, **kwargs)

  File "/tmp/tmpc8dT86/oozie/editor2/gen/workflow.xml.mako.py", line 110, in render_body
    __M_writer( node.to_xml(mapping, node_mapping, workflow_mapping) )

  File "/opt/hue/apps/oozie/src/oozie/models2.py", line 317, in to_xml
    return django_mako.render_to_string(self.get_template_name(), data)

  File "/opt/hue/desktop/core/src/desktop/lib/django_mako.py", line 114, in render_to_string_normal
    result = template.render(**data_dict)

  File "/opt/hue/build/env/lib/python2.6/site-packages/Mako-0.8.1-py2.6.egg/mako/template.py", line 443, in render
    return runtime._render(self, self.callable_, args, data)

  File "/opt/hue/build/env/lib/python2.6/site-packages/Mako-0.8.1-py2.6.egg/mako/runtime.py", line 786, in _render
    **_kwargs_for_callable(callable_, data))

  File "/opt/hue/build/env/lib/python2.6/site-packages/Mako-0.8.1-py2.6.egg/mako/runtime.py", line 818, in _render_context
    _exec_template(inherit, lclcontext, args=args, kwargs=kwargs)

  File "/opt/hue/build/env/lib/python2.6/site-packages/Mako-0.8.1-py2.6.egg/mako/runtime.py", line 844, in _exec_template
    callable_(context, *args, **kwargs)

  File "/tmp/tmpc8dT86/oozie/editor2/gen/workflow-join.xml.mako.py", line 31, in render_body
    __M_writer(escape(unicode( node_mapping[node['children'][0]['to']].name )))

IndexError: list index out of range

```

**Comment by searchwizard** [comment]
I see the error "IndexError: list index out of range" a lot. 

For example:
When "OK" transition points to a wrong node (when I add and delete an action/fork/join etc.., the names are still in the drop down)

**Comment by romainr** [comment]
I could repro it, will have a fix soon

**Comment by romainr** [comment]
https://issues.cloudera.org/browse/HUE-2712

**Comment by romainr** [comment]
Fixed in 3.8.1 and 3.9

**Comment by searchwizard** [comment]
Applied the patch in my system. It works great. Thanks
